lib/teamsnap.rb

```diff
@@ -0,0 +1,261 @@
+require "faraday"
+require "typhoeus"
+require "typhoeus/adapters/faraday"
+require "oj"
+require "inflecto"
+require "virtus"
+require "date"
+
+require_relative "teamsnap/version"
+
+Oj.default_options = {
+  :mode => :compat,
+  :symbol_keys => true,
+  :class_cache => true
+}
+
+Faraday::Request.register_middleware(
+  :teamsnap_auth_middleware => -> { TeamSnap::AuthMiddleware }
+)
+
+Inflecto.inflections do |inflect|
+  inflect.irregular "member_preferences", "members_preferences"
+  inflect.irregular "opponent_results", "opponents_results"
+  inflect.irregular "team_preferences", "teams_preferences"
+  inflect.irregular "team_results", "teams_results"
+end
+
+module TeamSnap
+  EXCLUDED_RELS = ["me", "apiv2_root", "root", "self"]
+  DEFAULT_URL = "http://localhost:3000"
+  Error = Class.new(StandardError)
+  NotFound = Class.new(TeamSnap::Error)
+
+  class AuthMiddleware < Faraday::Middleware
+    def initialize(app, options)
+      super(app)
+      @options = options
+    end
+
+    def call(env)
+      token = @options.fetch(:token)
+      env[:request_headers].merge!({"Authorization" => "Bearer #{token}"})
+
+      @app.call(env)
+    end
+  end
+
+  class << self
+    def collection(href, resp)
+      Module.new do
+        define_singleton_method(:included) do |descendant|
+          descendant.send(:include, TeamSnap::Item)
+          descendant.extend(TeamSnap::Collection)
+          descendant.instance_variable_set(:@href, href)
+          descendant.instance_variable_set(:@resp, resp)
+        end
+      end
+    end
+
+    def init(token, opts = {})
+      opts[:url] ||= DEFAULT_URL
+
+      self.client = Faraday.new(
+        :url => opts.fetch(:url),
```

Comment:
it looks like you are only using the `:url` here, you could pass it in as a default to the block instead of memoizing it above. I think there is merit in this approach, just putting it up as an option.